internal/vcr/retry.go

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package vcr
+
+import (
+	"errors"
+	"net/url"
+	"strings"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+)
+
+// InteractionNotFoundRetryableFunc is a retryable function to augment retry behavior for AWS service clients
+// when VCR testing is enabled
+var InteractionNotFoundRetryableFunc = func(err error) aws.Ternary {
+	var urlError *url.Error
+	if errors.As(err, &urlError) && strings.Contains(err.Error(), "requested interaction not found") {
```

Comment:
Could use `errs.IsA[*url.Error]`.

Reply:
Since `url.Error` does not implement an `ErrorMessage` method, I couldn't use `errs.IsAErrorMessageContains` and opted for the standard library since the `strings.Contains` bit will necessary anyway.

I can change to this if that reads more cleanly.

```go
	if errs.IsA[*url.Error](err) && strings.Contains(err.Error(), "requested interaction not found") {
		return aws.FalseTernary
	}
```

Reply:
Up to you 😄.